pkg/controller/update_cluster.go

```diff
@@ -105,6 +110,39 @@ func (c *M3DBController) createNamespaces(cluster *myspec.M3DBCluster, registry
 	return nil
 }
 
+// readyNamespaces will attempt to mark all namespaces in the initializing state as ready.
+func (c *M3DBController) readyNamespaces(cluster *myspec.M3DBCluster, registry *dbns.Registry) error {
+	// NB(nate): Marking namespaces ready (without force) involves checking dbnodes for namespace
+	// availability. External coordinators do not have connections to dbnodes so don't attempt this
+	// step for now.
+	if cluster.Spec.ExternalCoordinator != nil {
+		c.logger.Debug("using an external coordinator. will not attempt to ready namespaces")
+		return nil
+	}
+
+	toReady := namespacesToReady(registry, cluster.Spec.Namespaces)
+	for _, ns := range toReady {
+		req := &admin.NamespaceReadyRequest{
+			Name: ns.Name,
+		}
+		err := c.adminClient.namespaceClientForCluster(cluster).Ready(req)
+		if pkgerrors.Cause(err) == m3admin.ErrNotFound {
+			c.logger.Info("coordinator does not yet support the ready endpoint. " +
```

Comment:
This is probably valuable enough context to keep at `Info`, but just a heads up this will log every time a cluster is processed.